connect/json/src/test/java/org/apache/kafka/connect/json/JsonConverterTest.java

```diff
@@ -563,6 +563,14 @@ public void structToJson() {
                 converted.get(JsonSchema.ENVELOPE_PAYLOAD_FIELD_NAME));
     }
 
+    @Test
+    public void structSchemaIdentical() {
+        Schema schema = SchemaBuilder.struct().field("field1", Schema.BOOLEAN_SCHEMA).field("field2", Schema.STRING_SCHEMA).field("field3", Schema.STRING_SCHEMA).field("field4", Schema.BOOLEAN_SCHEMA).build();
+        Schema inputSchema = SchemaBuilder.struct().field("field1", Schema.BOOLEAN_SCHEMA).field("field2", Schema.STRING_SCHEMA).field("field3", Schema.STRING_SCHEMA).field("field4", Schema.BOOLEAN_SCHEMA).build();
+        Struct input = new Struct(inputSchema).put("field1", true).put("field2", "string2").put("field3", "string3").put("field4", false);
+        assertTrue(convertStructEqual(schema, input));
```

Comment:
Would be nice to break these into multiple lines so it is more readable and easier to discern the structure of each `Schema`.

Also, re @bartektartanus's comments, it'd be nice to add another `structSchemaSame` test that verifies this works when the same schema instance is used. Yes, it "should" work, but let's add a test to prove it.

Finally, and this is really minor and probably more of a preference, but you might consider changing the `convertStructEqual` method into an `assertStructSchemaEqual` method that actually performs the assertion. As a result, that method could be a bit simpler _and_ line 571 would be simpler:

    assertStructSchemaEqual(schema, input)